dtfs-central-api/api-tests/v1/utilisation-reports/get-utilisation-reports.api-test.ts

```diff
@@ -1,61 +1,123 @@
 import { Response } from 'supertest';
-import { MONGO_DB_COLLECTIONS, UtilisationReport } from '@ukef/dtfs2-common';
-import wipeDB from '../../wipeDB';
+import { AzureFileInfo, IsoDateTimeStamp, ReportPeriod, UtilisationReportEntity, UtilisationReportReconciliationStatus } from '@ukef/dtfs2-common';
+import { SqlDbDataSource } from '@ukef/dtfs2-common/sql-db-connection';
+import axios from 'axios';
+import {
+  aNonUploadedMarkedReconciledUtilisationReportEntity,
+  aNotReceivedUtilisationReportEntity,
+  anUploadedUtilisationReportEntity,
+} from '../../mocks/entities/utilisation-report-entity';
 import app from '../../../src/createApp';
-import db from '../../../src/drivers/db-client';
-import { MOCK_UTILISATION_REPORT } from '../../mocks/utilisation-reports/utilisation-reports';
 import createApi from '../../api';
+import { wipeAllUtilisationReports } from '../../test-helpers/wipe-sql-db';
 
 const api = createApi(app);
 
 const getUrl = (bankId: string) => `/v1/bank/${bankId}/utilisation-reports`;
 
+const saveReportToDatabase = async (report: UtilisationReportEntity): Promise<UtilisationReportEntity> =>
+  await SqlDbDataSource.getRepository(UtilisationReportEntity).save(report);
+
+type UtilisationReportResponseBody = {
+  id: number;
+  bankId: string;
+  status: UtilisationReportReconciliationStatus;
+  uploadedByUserId: string;
+  reportPeriod: ReportPeriod;
+  azureFileInfo: AzureFileInfo | null;
+  dateUploaded: IsoDateTimeStamp;
+};
+
 interface CustomResponse extends Response {
   body:
     | {
         errors: {
           msg: string;
         }[];
       }
-    | ({ errors?: undefined } & UtilisationReport);
+    | ({ errors?: undefined } & UtilisationReportResponseBody[]);
```

Comment:
I think we probably want to change this type now seeing as we're losing its benefits by needing all the `as` type assertions below. My idea would be that we have two functions which assert type correct type and two response types: a `success` and `error` response:
```typescript
interface CustomErrorResponse extends Response {
  body: {
    errors: {  msg: string }[];
  };
}

interface CustomResponse extends Response {
  body: UtilisationReportResponseBody;
}

// inside `describe` - needs to use `function` instead of `() => {}` to use `asserts <> is <>`
function assertIsErrorResponse(response: Response): asserts response is CustomErrorResponse {
  if (!('errors' in response.body)) {
    throw new Error("Expected error response to contain 'response.body.errors'");
  }
  expect(Array.isArray(response.body.errors)).toBe(true);
  response.body.errors.forEach((error) => {
    if (!('msg' in error)) {
      throw new Error("Expected 'response.body.errors' array items to all contain 'msg' property");
    }
    expect(typeof error.msg === 'string').toBe(true);
  );
}
```
with a similar function for `assertIsSuccessResponse`. This would give us the type safety we want as well as giving us some testing on what the endpoint returns for free

Reply:
Ok so I have gone with the two types as you suggest which I agree is much nicer, but then I have perhaps been a bit sneaky and just typed the response as whichever one we expect it to be. I think this is ok, because assertions will fail anyway if it returns the incorrect type, and the status type is in some respects doing the type check for us. It also prevents us having to write complex type checking functions which I'm not sure are generally worth the effort? Happy to discuss further if you do think this is indeed a bit sneaky, I'm just keen to keep tests as easy to write as possible!